app/src/main/java/at/roteskreuz/stopcorona/model/managers/ChangelogManager.kt

```diff
@@ -55,29 +67,39 @@ class ChangelogManagerImpl(
 
     private var lastSeenChangelogId: Int by preferences.intSharedPreferencesProperty(PREF_LAST_SEEN_CHANGELOG_ID, 0)
 
-    private val hasBeenDisplayed: Boolean
-        get() = changelog.id <= lastSeenChangelogId
-
     override fun unseenChangelogForVersionAvailable(version: String): Boolean {
-        return changelog.versions.contains(convertVersionName(version)) && hasBeenDisplayed.not()
+        return changelog.versions.contains(convertVersionName(version)) && hasBeenDisplayed(lastSeenChangelogId).not()
     }
 
     override fun getChangelogForVersion(version: String): Changelog? {
         return if (unseenChangelogForVersionAvailable(convertVersionName(version))) {
-            lastSeenChangelogId = changelog.id
             changelog
         } else {
             null
         }
     }
 
+    override fun observeExposureSDKReadyToStart(): Observable<Boolean> {
+        return preferences.observeInt(PREF_LAST_SEEN_CHANGELOG_ID, 0).map {
+            hasBeenDisplayed(it)
+        }
+    }
+
+    override fun flagChangelogAsSeen() {
```

Comment:
Suggestion: Rename to `markChangelogAsSeen`

Reply:
done